components/org.wso2.finance.open.banking.conformance.mgt/src/main/java/org/wso2/finance/open/banking/conformance/mgt/dao/impl/ReportDAOImpl.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.finance.open.banking.conformance.mgt.dao.impl;
+
+import com.google.gson.Gson;
+import org.wso2.finance.open.banking.conformance.mgt.dao.ReportDAO;
+import org.wso2.finance.open.banking.conformance.mgt.db.DBConnector;
+import org.wso2.finance.open.banking.conformance.mgt.db.SQLConstants;
+import org.wso2.finance.open.banking.conformance.mgt.models.Report;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implementation of the ReportDAO
+ * Responsible for the database operations regarding reports
+ */
+public class ReportDAOImpl implements ReportDAO {
+
```

Comment:
In general the cyclomatic complexity of following methods seems to be high and see if there is a way to refactor these methods and make them simple.
storeReport
getNewReportID
getReport
getReports

Reply:
done. used jdbc template.